### ISSUE TITLE
Probability is mismatching  for the image classification example

### ISSUE THREAD

**Issue by spadiyar67** [tensorflow/tensorflow#49893]
used same code what given in the below url ,
https://www.tensorflow.org/tutorials/images/classification

created the .tflite for Android and imported the tflite but while executing the code Probability is not matching with the python code.

Percentage in python is ,
This image most likely belongs to sunflowers with a **95.65** percent confidence. 

The android code:
val model = Model.newInstance(this)
        val bitmap: Bitmap =
            BitmapFactory.decodeResource(resources, R.drawable.image)
// Creates inputs for reference.
        val image = TensorImage.fromBitmap(bitmap)

// Runs model inference and gets result.
        val outputs = model.process(image)
        val probability = outputs.probabilityAsCategoryList

// Releases model resources if no longer used.
        model.close()

        val outputsProbability = probability.apply {
            sortByDescending { it.score } // Sort with highest confidence first
        }.take(5)

        val items = mutableListOf<Recognition>()
        for (output in outputsProbability) {
            items.add(Recognition(output.label, output.score))
        }

        Log.d("propability", items.toString())
        
o/p:
D/propability: [dandelion / 554.4%, daisy / 61.7%, roses / -75.3%, tulips / -282.9%, sunflowers / **-426.3**%]

**Comment by tilakrayal** [comment]
@spadiyar67 ,

We see that the issue [template](https://github.com/tensorflow/tensorflow/issues/new/choose) has not been filled, could you please do so as it helps us analyse the issue [tf version, steps followed before you ran into this error or if possible share a colab gist with the issue reported.

Thanks!

**Comment by spadiyar67** [comment]
@tilakrayal 
The Tf version is 2.4.1 


I have given python and android code below.
Steps:
Created tflite from the python code ->imported in Android studio->android code i have given below->the probability while running python and android code mismatching.

**Python code**:

import matplotlib.pyplot as plt
import numpy as np
import os
import PIL
import tensorflow as tf
import matplotlib.pyplot as plt

from tensorflow import keras
from tensorflow.keras import layers
from tensorflow.keras.models import Sequential
from tflite_support import flatbuffers
from tflite_support import metadata as _metadata
from tflite_support import metadata_schema_py_generated as _metadata_fb
from absl import flags
import imageio
import cv2
import numpy as np
import pathlib


batch_size = 32
img_height = 180
img_width = 180
data_dir = ".keras/datasets/flower_photos"

train_ds = tf.keras.preprocessing.image_dataset_from_directory(
    data_dir,
  validation_split=0.2,
  subset="training",
  seed=123,
  image_size=(img_height, img_width),
  batch_size=batch_size)

val_ds = tf.keras.preprocessing.image_dataset_from_directory(
  data_dir,
  validation_split=0.2,
  subset="validation",
  seed=123,
  image_size=(img_height, img_width),
  batch_size=batch_size)

class_names = train_ds.class_names
print(class_names)



for image_batch, labels_batch in train_ds:
  print(image_batch.shape)
  print(labels_batch.shape)
  break

data_augmentation = keras.Sequential(
  [
    layers.experimental.preprocessing.RandomFlip("horizontal",
                                                 input_shape=(img_height,
                                                              img_width,
                                                              3)),
    layers.experimental.preprocessing.RandomRotation(0.1),
    layers.experimental.preprocessing.RandomZoom(0.1),
  ]
)
num_classes = 5
model = Sequential([
  data_augmentation,
  layers.experimental.preprocessing.Rescaling(1./255),
  layers.Conv2D(16, 3, padding='same', activation='relu'),
  layers.MaxPooling2D(),
  layers.Conv2D(32, 3, padding='same', activation='relu'),
  layers.MaxPooling2D(),
  layers.Conv2D(64, 3, padding='same', activation='relu'),
  layers.MaxPooling2D(),
  layers.Dropout(0.2),
  layers.Flatten(),
  layers.Dense(128, activation='relu'),
  layers.Dense(num_classes)
])

model.compile(optimizer='adam',
              loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
              metrics=['accuracy'])


print(len(train_ds))
print(len(val_ds))

epochs = 15
history = model.fit(
  train_ds,
  validation_data=val_ds,
  epochs=epochs
)
print(len(train_ds))
print(len(val_ds))


sunflower_url = "https://storage.googleapis.com/download.tensorflow.org/example_images/592px-Red_sunflower.jpg"
sunflower_path = tf.keras.utils.get_file('Red_sunflower', origin=sunflower_url)

img = keras.preprocessing.image.load_img(
    sunflower_path, target_size=(img_height, img_width)
)
img_array = keras.preprocessing.image.img_to_array(img)
img_array = tf.expand_dims(img_array, 0) # Create a batch

predictions = model.predict(img_array)
score = tf.nn.softmax(predictions[0])

print(
    "This image most likely belongs to {} with a {:.2f} percent confidence."
    .format(class_names[np.argmax(score)], 100 * np.max(score))
)

# Convert the model.
converter = tf.lite.TFLiteConverter.from_keras_model(model)
tflite_model = converter.convert()

# Save the model.
with open('model.tflite', 'wb') as f:
  f.write(tflite_model)



""" ... """
"""Creates the metadata for an image classifier."""

# Creates model info.
model_meta = _metadata_fb.ModelMetadataT()
model_meta.name = "MobileNetV1 image classifier"
model_meta.description = ("Identify the most prominent object in the "
                          "image from a set of 1,001 categories such as "
                          "trees, animals, food, vehicles, person etc.")
model_meta.version = "v1"
model_meta.author = "TensorFlow"
model_meta.license = ("Apache License. Version 2.0 "
                      "http://www.apache.org/licenses/LICENSE-2.0.")

# Creates input info.
input_meta = _metadata_fb.TensorMetadataT()
# Creates output info.
output_meta = _metadata_fb.TensorMetadataT()

input_meta.name = "image"
input_meta.description = (
    "Input image to be classified. The expected image is {0} x {1}, with "
    "three channels (red, blue, and green) per pixel. Each value in the "
    "tensor is a single byte between 0 and 255.".format(160, 160))
input_meta.content = _metadata_fb.ContentT()
input_meta.content.contentProperties = _metadata_fb.ImagePropertiesT()
input_meta.content.contentProperties.colorSpace = (
    _metadata_fb.ColorSpaceType.RGB)
input_meta.content.contentPropertiesType = (
    _metadata_fb.ContentProperties.ImageProperties)
input_normalization = _metadata_fb.ProcessUnitT()
input_normalization.optionsType = (
    _metadata_fb.ProcessUnitOptions.NormalizationOptions)
input_normalization.options = _metadata_fb.NormalizationOptionsT()
input_normalization.options.mean = [127.5]
input_normalization.options.std = [127.5]
input_meta.processUnits = [input_normalization]
input_stats = _metadata_fb.StatsT()
input_stats.max = [255]
input_stats.min = [0]
input_meta.stats = input_stats

# Creates output info.
output_meta = _metadata_fb.TensorMetadataT()
output_meta.name = "probability"
output_meta.description = "Probabilities of the 1001 labels respectively."
output_meta.content = _metadata_fb.ContentT()
output_meta.content.content_properties = _metadata_fb.FeaturePropertiesT()
output_meta.content.contentPropertiesType = (
    _metadata_fb.ContentProperties.FeatureProperties)
output_stats = _metadata_fb.StatsT()
output_stats.max = [1.0]
output_stats.min = [0.0]
output_meta.stats = output_stats
label_file = _metadata_fb.AssociatedFileT()
label_file.name = os.path.basename("/MachineLearning/TensorFlow/Label.txt")
label_file.description = "Labels for objects that the model can recognize."
label_file.type = _metadata_fb.AssociatedFileType.TENSOR_AXIS_LABELS
output_meta.associatedFiles = [label_file]

# Creates subgraph info.
subgraph = _metadata_fb.SubGraphMetadataT()
subgraph.inputTensorMetadata = [input_meta]
subgraph.outputTensorMetadata = [output_meta]
model_meta.subgraphMetadata = [subgraph]

b = flatbuffers.Builder(0)
b.Finish(
    model_meta.Pack(b),
    _metadata.MetadataPopulator.METADATA_FILE_IDENTIFIER)
metadata_buf = b.Output()

populator = _metadata.MetadataPopulator.with_model_file("model.tflite")
populator.load_metadata_buffer(metadata_buf)
populator.load_associated_files(["/MachineLearning/TensorFlow/Label.txt"])
populator.populate()

Android Code:
val model = Model.newInstance(this)
val bitmap: Bitmap =
BitmapFactory.decodeResource(resources, R.drawable.image)
// Creates inputs for reference.
val image = TensorImage.fromBitmap(bitmap)

// Runs model inference and gets result.
val outputs = model.process(image)
val probability = outputs.probabilityAsCategoryList

// Releases model resources if no longer used.
model.close()

 val outputsProbability = probability.apply {
        sortByDescending { it.score } // Sort with highest confidence first
    }.take(5)

    val items = mutableListOf<Recognition>()
    for (output in outputsProbability) {
        items.add(Recognition(output.label, output.score))
    }

    Log.d("propability", items.toString())

**Comment by tilakrayal** [comment]
@spadiyar67 ,

On executing the given code, I am facing an error. Please find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/0dcae8fbfd711e9dc58213b703de4166/49893.ipynb) and share all dependencies to replicate the issue or share a colab gist with the reported error.

Thanks!

**Comment by spadiyar67** [comment]
@tilakrayal 
I have uploaded all the necessary files and now able to create the tflite file in the colab . could you please give a try and check the probability in both the platform.

If you see same error  then create "Label.txt" file and add  below labels inside the file .
daisy
dandelion
roses
sunflowers
tulips

**Comment by tilakrayal** [comment]
@spadiyar67 ,

The code provided is fairly complex and indentation errors hence it would be difficult for us to pinpoint the issue. Could you please get the example down to the simplest possible repro?  Please share the colab link.That will allow us to determine the source of the issue easily. Thanks!
